sdk/appconfiguration/azure-data-appconfiguration/inc/azure/data/appconfiguration/configuration_client_models.hpp

```diff
@@ -0,0 +1,614 @@
+    static UpdateSnapshotResponseContentType const ApplicationProblemJson;
+  };
+
+  class UpdateSnapshotRequestContentType final
+      : public Core::_internal::ExtendableEnumeration<UpdateSnapshotRequestContentType> {
+  public:
+    UpdateSnapshotRequestContentType() = default;
+
+    explicit UpdateSnapshotRequestContentType(std::string updateSnapshotRequestContentType)
+        : ExtendableEnumeration(std::move(updateSnapshotRequestContentType))
+    {
+    }
+
+    static UpdateSnapshotRequestContentType const ApplicationMergePatchJson;
+    static UpdateSnapshotRequestContentType const ApplicationJson;
+  };
+
+  class GetLabelsResponseContentType final
+      : public Core::_internal::ExtendableEnumeration<GetLabelsResponseContentType> {
+  public:
+    GetLabelsResponseContentType() = default;
+
+    explicit GetLabelsResponseContentType(std::string getLabelsResponseContentType)
+        : ExtendableEnumeration(std::move(getLabelsResponseContentType))
+    {
+    }
+
+    static GetLabelsResponseContentType const ApplicationVndMicrosoftAppconfigLabelsetJson;
+    static GetLabelsResponseContentType const ApplicationProblemJson;
+  };
+
+  class LabelFields final : public Core::_internal::ExtendableEnumeration<LabelFields> {
+  public:
+    LabelFields() = default;
+
+    explicit LabelFields(std::string labelFields) : ExtendableEnumeration(std::move(labelFields)) {}
+
+    static LabelFields const Name;
+  };
+
+  class GetRevisionsResponseContentType final
+      : public Core::_internal::ExtendableEnumeration<GetRevisionsResponseContentType> {
+  public:
+    GetRevisionsResponseContentType() = default;
+
+    explicit GetRevisionsResponseContentType(std::string getRevisionsResponseContentType)
+        : ExtendableEnumeration(std::move(getRevisionsResponseContentType))
+    {
+    }
+
+    static GetRevisionsResponseContentType const ApplicationVndMicrosoftAppconfigKvsetJson;
+    static GetRevisionsResponseContentType const ApplicationProblemJson;
+  };
+
+  struct Key final
+  {
+    std::string Name;
+  };
+
+  struct CheckKeysResult final
+  {
+    std::string SyncToken;
+  };
+
+  struct KeyValue final
+  {
+    std::string Key;
+    Nullable<std::string> Label;
+    Nullable<std::string> ContentType;
+    Nullable<std::string> Value;
+    Nullable<std::string> LastModified;
+    Nullable<std::map<std::string, std::string>> Tags;
+    Nullable<bool> Locked;
+    Nullable<std::string> Etag;
+  };
+
+  struct CheckKeyValuesResult final
+  {
+    std::string SyncToken;
+    std::string ETag;
+  };
+
+  struct GetKeyValueResult final
+  {
+    std::string Key;
+    Nullable<std::string> Label;
+    Nullable<std::string> ContentType;
+    Nullable<std::string> Value;
+    Nullable<std::string> LastModified;
+    Nullable<std::map<std::string, std::string>> Tags;
+    Nullable<bool> Locked;
+    Nullable<std::string> Etag;
+    Core::Uuid XMsClientRequestId{};
+    std::string SyncToken;
```

Comment:
This might be an emitter issue, since the syncTokenHeader model contains an optional string in the tsp:
https://github.com/Azure/azure-rest-api-specs/blob/7dfd17073198d5cf3df99a106b770cb01b395f2a/specification/appconfiguration/AppConfiguration/routes.tsp#L29-L33